lib/internal/modules/cjs/loader.js

```diff
@@ -599,9 +608,12 @@ Module._findPath = function(request, paths, isMain) {
 
   // For each path
   for (let i = 0; i < paths.length; i++) {
-    // Don't search further if path doesn't exist and request is inside the path
+    // Don't search further if path doesn't exist
+    // or doesn't have permission to it
     const curPath = paths[i];
-    if (insidePath && curPath && _stat(curPath) < 1) continue;
+    if (insidePath && curPath &&
+      (permission?.has('fs.read', curPath) === false || _stat(curPath) < 1)
+    ) continue;
```

Comment:
Sorry if the indentation is off in this suggestion, but please apply curly braces here.

```suggestion
    ) {
    continue;
  }
```